hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/stats/ColumnStatsIndices.java

```diff
@@ -85,16 +89,109 @@ public class ColumnStatsIndices {
   private ColumnStatsIndices() {
   }
 
-  public static List<RowData> readColumnStatsIndex(String basePath, HoodieMetadataConfig metadataConfig, String[] targetColumns) {
+  public static Set<String> candidatePartitionsInMetadataTable(
```

Comment:
Can we have some basic abstractions so that it is more easy for extention, for e.g. the RLI support in the near future.